### DISCUSSION TITLE
Resizing or scaling removes RGB data in fully transparent pixels. How can I preserve this data?

### DISCUSSION THREAD

**Discussion by frugbug** [ImageMagick/ImageMagick#4562] [Help] [answered]
The command I am using: `magick convert input.png -virtual-pixel edge -scale 64x64 output.png`
I've also tried using `-resize` with no success.
I need to preserve the data in fully transparent pixels but IM removes it, replacing it with black. How can I avoid this behaviour and instead preserve the information in these pixels?

Example before, with alpha channel: 
![test](https://user-images.githubusercontent.com/67615785/145147255-da09282d-cf80-4f57-9dc4-8fc2bff2368d.png)
Example after, with alpha channel:
![test](https://user-images.githubusercontent.com/67615785/145147175-ea466a28-a8d4-4e50-b996-353e37617d04.png)
Example before, no alpha:
![image](https://user-images.githubusercontent.com/67615785/145147313-96bfcb2c-12a5-42c0-b0a8-0dfd716bf8f2.png)
Example after, no alpha:
![image](https://user-images.githubusercontent.com/67615785/145147361-b77c34a5-fac5-40e2-a4b7-6018e97f00d1.png)

Thanks in advance for any help

**Comment by GeeMack** [comment]
To preserve transparency it's good practice to specify a transparent background...

    magick input.png -background none -virtual-pixel edge -scale 64x64 output.png

Also, since you're using IMv7, unless you have reason to use ImageMagick v6 behavior, you should omit the `convert` and just use the command `magick [...]`.

**Reply by frugbug** [reply]
Thank you but unfortunately specifying background doesn't work, I get the same result.

**Reply by GeeMack** [reply]
Try adding `-alpha set` right after reading in the input image.

**Reply by frugbug** [reply]
Still doesn't work

**Comment by fmw42** [comment]
try

`magick 145147255-da09282d-cf80-4f57-9dc4-8fc2bff2368d.png -background none -virtual-pixel none -scale 64x64 PNG32:result.png
`

![result](https://user-images.githubusercontent.com/13458218/145156953-23b87b52-0f32-4288-8e6c-d219ce903a46.png)

**Reply by frugbug** [reply]
Doesn't work, transparent pixels are still black

**Reply by fmw42** [reply]
Works fine for me as above. Is not my thumbnail transparent?  If not, what is your ImageMagick version and platform and what viewer are you using?

**Reply by snibgo** [reply]
FlatJim doesn't want transparent black, but transparent some-other-colour, where the colour is what would have resulted from downsizing the input if it had been fully opaque.

**Comment by snibgo** [comment, marked as answer]
I suggest using just "magick", not "magick convert".

When upsizing or downsizing, the colour and alpha of each output pixel is calculated from the colour and alpha of a number of input pixels. The weight of each input pixel depends on its alpha, so an input pixel that is fully transparent carries no weight in the calculation.

Hence an output pixel that is calculated from pixels that are all fully transparent will be fully transparent but with no colour at all. [EDIT: Strictly speaking, the colour is undefined. It cannot be calculated.]

To get what you want, you need to resize the colour image with no alpha. Also resize the alpha channel, and copy it back in.

Windows BAT syntax:
```
magick ^
  14alph.png ^
  ( +clone -alpha extract ) ^
  -alpha off ^
  -resize 100x100 ^
  -compose CopyOpacity -composite ^
  out.png
```
Note that "-resize" operates on all the images in the current list. There are two images: the colour with no alpha, and the extracted alpha.

**Reply by frugbug** [reply]
Thank you so much! And thanks for taking the time to write the explanation too, it was really helpful
